electroncash/tor/controller.py

```diff
@@ -117,10 +117,23 @@ def __del__(self):
         self.status_changed.clear()
         self.active_port_changed.clear()
 
-    # Opened Socks listener on 127.0.0.1:53544
-    # Opened Control listener on 127.0.0.1:3300
-    _listener_re = re.compile(r".*\[notice\] Opened ([^ ]*) listener on (.*)$")
-    _endpoint_re = re.compile(r".*:(\d*)")
+    # Version 0.4.5.5
+    # [notice] Opening Socks listener on 127.0.0.1:0
+    # [notice] Socks listener listening on port 36103.
+    # [notice] Opened Socks listener connection (ready) on 127.0.0.1:36103
+    # [notice] Opening Control listener on 127.0.0.1:0
+    # [notice] Control listener listening on port 36104.
+    # [notice] Opened Control listener connection (ready) on 127.0.0.1:36104
+
+    # Version 0.4.2.5
+    # [notice] Opening Socks listener on 127.0.0.1:0
+    # [notice] Socks listener listening on port 36103.
+    # [notice] Opened Socks listener on 127.0.0.1:36103
+    # [notice] Opening Control listener on 127.0.0.1:0
+    # [notice] Control listener listening on port 36104.
+    # [notice] Opened Control listener on 127.0.0.1:36104
+
+    _listener_re = re.compile(r".*\[notice\] ([^ ]*) listener listening on port ([0-9]*)\.?$")
```

Comment:
Nit: Shouldn't the port match be `[0-9]+`?  Otherwise you may get a match but then an exception when you cast it to `int()` below... (empty string matches this).

I realize it doesn't matter in practice but.. just a nit.

Reply:
You're right, that's a good point

Reply:
Thanks